src/Core/MySQL/PacketsProtocolBinary.h

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include <vector>
+#include <Columns/IColumn.h>
+#include <Core/MySQL/IMySQLReadPacket.h>
+#include <Core/MySQL/IMySQLWritePacket.h>
+#include "DataTypes/IDataType.h"
+#include "DataTypes/Serializations/ISerialization.h"
+
+namespace DB
+{
+
+namespace MySQLProtocol
+{
+
+    namespace ProtocolBinary
+    {
+        class ResultSetRow : public IMySQLWritePacket
+        {
+        private:
+            TypeIndex getTypeIndex(DataTypePtr data_type, const ColumnPtr & col) const;
+
+        protected:
+            int row_num;
+            const Columns & columns;
+            const DataTypes & data_types;
+            const Serializations & serializations;
+
+            std::vector<String> serialized = std::vector<String>(columns.size());
+
+            size_t null_bitmap_size = (columns.size() + 7 + 2) / 8;
```

Comment:
It is not exactly clear where the constants come from. Maybe add a comment or a reference to the MySQL protocol docs.

Reply:
And should a "null bitmap" not have a NULL entry per row? So one would expect that the size depends on sth like `columns[0]->size()`.

EDIT: Perhaps it is because the class describes a single *row*? In that case, `null_bitmap` is a strange name. Maybe just `null_columns`.

Reply:
`null_bitmap` is just how it's named in [the doc](https://dev.mysql.com/doc/dev/mysql-server/latest/page_protocol_binary_resultset.html#sect_protocol_binary_resultset_row). I think we will introduce even more confusion if we change the name.